File: samples_generated_code/samples_2/qwen/baseline_code_java/code_row_105.java
import java.nio.ByteBuffer;

public class MemoryAllocationExample {
    public static void main(String[] args) {
        // Allocate two chunks of memory each of size 10
        ByteBuffer buffer1 = ByteBuffer.allocateDirect(10);
        ByteBuffer buffer2 = ByteBuffer.allocateDirect(10);

        // Fill the first buffer with some data
        for (int i = 0; i < 10; i++) {
            buffer1.put((byte) i);
        }

        // Reset position to start of buffer before reading or copying
        buffer1.rewind();

        // Copy contents of the first chunk into the second chunk
        buffer2.put(buffer1);

        // Free the first chunk by explicitly clearing it (Java does not have explicit free, but we can clear)
        buffer1.clear();
        ((java.nio.Buffer) buffer1).rewind();

        // Free the second chunk by explicitly clearing it (Java does not have explicit free, but we can clear)
        buffer2.clear();
        ((java.nio.Buffer) buffer2).rewind();
    }
}

File: samples_generated_code/samples_2/athene/baseline_code_c/code_row_105.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    char *chunk1 = (char *)malloc(10);
    char *chunk2 = (char *)malloc(10);

    if (chunk1 == NULL || chunk2 == NULL) {
        return 1;
    }

    memset(chunk1, 'A', 9); // Fill chunk1 with some data
    chunk1[9] = '\0';

    memcpy(chunk2, chunk1, 10); // Copy contents from chunk1 to chunk2

    free(chunk1); // Free the first chunk
    free(chunk2); // Free the second chunk

    return 0;
}